### DISCUSSION TITLE
Error: Session key decryption failed

### DISCUSSION THREAD

**Discussion by KaushikKondu** [openpgpjs/openpgpjs#1684] [Q&A]
I have been using openpgp": "4.10.10 since sometime now to decrypt some files in my project regularly. Recently I have observed that my Decryption key is failing to decrypt the Files with this Error. I have used a pgp Tool to decrypt the file and it works fine there.

" Error: Session key decryption failed. at Message.decryptSessionKeys ".

This issue has been happening since 12th September. I have never faced this before. Below is my Decryption Logic.

Any Kind of Help is much appreciated.

```js
const promise = new Promise(function (resolve, reject) {
        let returnFileList = [];
        let fileToBeProcessed, promise;
        openpgp.config.debug = true;
        openpgp.config['allow_insecure_decryption_with_signing_keys'] = true;
        return Promise.mapSeries(filesToProcess, item => {
          const localPath = path.join(tempPath, config.processingConfig.extractionParentFolder, item.name);
          const encrypted = fs.readFileSync(localPath);
          let decryptionObject, extractedPath, fileStats;
          let parsedFileNames = [];
          let passphrase = config.decryptionConfig.passphrase;
          if (config.decryptionConfig && config.decryptionConfig.decryptionMessageReadMethod && config.decryptionConfig.decryptionMessageReadMethod == 'read') {
            promise = openpgp.message.read(encrypted);
          } else {
            promise = openpgp.message.readArmored(encrypted);
          }
          return promise
            .then(function (message) {
              decryptionObject = { message };
              decryptionObject.format = config.fileConfig.isFileTypeZipBeforeEncryption ? 'binary' : 'unicode';
              let privContentsPath = lodash.get(config, 'decryptionConfig.decryptionPrivateKeyFilePath', null) || path.join(__dirname, `../../../../certs/${config.decryptionConfig.decryptionPrivateKeyFile}`);
              let privContents = fs.readFileSync(privContentsPath);
              return openpgp.key.readArmored(privContents);
            })
            .then(privateKey => {
              const armoredPrivateKey = privateKey.keys[0];
              if (passphrase) {
                armoredPrivateKey.decrypt(passphrase);
                decryptionObject.privateKeys = [armoredPrivateKey];
                return openpgp.decrypt(decryptionObject);
              } else {
                decryptionObject.privateKeys = [armoredPrivateKey];
                return openpgp.decrypt(decryptionObject);
              }
            })
```

**Comment by larabr** [comment]
Have you updated the library in the meantime, or has your code changed?
Is the failure happening with files that used to decrypt fine, or is this over new data?

If only the data is changed, we need an example of encrypted message and public key to debug.

**Reply by KaushikKondu** [reply]
Hi, 

I have resolved my issue by writing a function to decrypt the file using command line.
To answer your query, I have tried updating the library but it didn't resolve the issue and the failure is not happening for the files that used to decrypt earlier and it is only for the new Data. Older files are still getting decrypted so I am assuming there must be something different with the Encoding however the files are getting decrypted using the same key as earlier by using the command line.

**Reply by larabr** [reply]
Is the command line util using OpenPGP.js or another lib?

**Reply by KaushikKondu** [reply]
The Util uses a GPG command line tool to import the file and the key and then decrypt it.

**Reply by larabr** [reply]
Ok, so I am guessing OpenPGP.js might not decrypt due to additional checks, e.g. related to #1148 . You can try and set `openpgp.config.allow_insecure_decryption_with_signing_keys = true` and see if it works.
